include/inexor/vulkan-renderer/world/cube.hpp

```diff
@@ -143,6 +151,11 @@ class Cube : public std::enable_shared_from_this<Cube> {
     /// Recursive way to collect all the caches.
     /// @param update_invalid If true it will update invalid polygon caches.
     [[nodiscard]] std::vector<PolygonCache> polygons(bool update_invalid = false) const;
+
+    /// Get the neighbor of this cube.
+    /// @param axis The axis on which to get the neighboring cube
+    /// @param direction Whether to get the cube which is above or below this cube on the selected axis
+    [[nodiscard]] std::shared_ptr<Cube> neighbor(NeighborAxis axis, NeighborDirection direction);
```

Comment:
Not sure if a description how the algorithm works would be nice here.

Reply:
I added a reference to a paper where I think it's better described than I ever could (and also to not bloat the comments here).

Reply:
Maybe you can cite aka. copy? There's a high chance that papers will disappear from the internet from one day to another.

Reply:
I think it would be way too much to copy and I think the comments in the code are sufficient for in-code documentation. I changed the link to an archive.org-version so it should be available in the foreseeable future.